airflow/config_templates/config.yml

```
@@ -1794,6 +1794,15 @@ webserver:
       type: string
       example: "sha256"
       default: "md5"
+    enable_test_connection:
+      description: |
+        The ability to allow testing connections from UI. By default, it is disabled.
+        If you're enabling it by setting it to True, make sure that you trust your users
+        who can add or edit connections in the UI and that they do not use it maliciously.
```

Another suggested wording, feel free to use or not (if used here, be sure to also copy it below in default config)
```suggestion
        Before setting this to True, make sure that you review the users who are able
        to add/edit connections in the UI and ensure they are trusted.
        Connection testing can can be done maliciously leading to undesired and insecure outcomes.
```